app/prepare_proposal.go

```diff
@@ -22,7 +22,10 @@ func (app *App) PrepareProposal(req abci.RequestPrepareProposal) abci.ResponsePr
 
 	// estimate the square size. This estimation errors on the side of larger
 	// squares but can only return values within the min and max square size.
-	squareSize, totalSharesUsed := estimateSquareSize(parsedTxs)
+	ctx := app.NewContext(true, core.Header{Height: app.LastBlockHeight()})
```

Comment:
That was discussed asynchronously, I will mark it resolved

Reply:
Out of curiosity what was the outcome of the discussion? Is a gas meter not needed because it wouldn't impact the usage of this context in this scope?